deployments.go

```diff
@@ -37,7 +37,7 @@ type DeploymentsCreateOpts struct {
 // deployer is an interface that represents something that can perform a
 // deployment.
 type deployer interface {
-	Deploy(context.Context, DeploymentsCreateOpts) (*Release, error)
+	Deploy(context.Context, *DeploymentsCreateOpts) (*Release, error)
```

Comment:
Any reason to make this a pointer? In general for option structs like this, I think it's better to not use a pointer unless the nil value has a semantic meaning. I think passing in nil right now would cause a panic.